src/node/blockstorage.cpp

```diff
@@ -648,8 +648,8 @@ bool BlockManager::FindBlockPos(FlatFilePos& pos, unsigned int nAddSize, unsigne
     if ((int)nFile != m_last_blockfile) {
         if (!fKnown) {
             LogPrint(BCLog::BLOCKSTORE, "Leaving block file %i: %s\n", m_last_blockfile, m_blockfile_info[m_last_blockfile].ToString());
+            FlushBlockFile(/*fFinalize=*/true, finalize_undo);
```

Comment:
Won't this result in the undo files not getting flushed when expected?

Reply:
See https://github.com/bitcoin/bitcoin/pull/27039#pullrequestreview-1320217718 for my thoughts on this,  a similar explanation was added to the OP.